src/ngraph/ngraph_emitter.cc

```diff
@@ -1285,6 +1286,119 @@ void Emitter::CreateLayerOps() {
           data, ngraph::AxisSet{static_cast<size_t>(seq_axis)});
     }
   };
+  ngraph_op_funcs_["SoftmaxOutput"] = [this](const NodePtr& node) {
+    auto input = op_map_[node->inputs_[0]];
+    auto in_shape = input->get_shape();
+
+    ngraph::AxisSet axes;
+    if (get_default(node, "multi_output", false)) {
+      axes.insert(1);
+    } else if (get_default(node, "preserve_shape", false)) {
+      axes.insert(in_shape.size() - 1);
+    } else {
+      for (size_t i = 1; i < in_shape.size(); ++i) {
```

Comment:
Would it make sense to replace this for-loop with something like the following?
``` c++
axes = pyrange(in_shape.size());
```

Reply:
I tried that, I could, but I'd have to create a pyrange that returned a set instead of a vector, AxisSet doesn't have a vector based constructor.

Reply:
We might be near the threshold of creating a utility function that does this.